### ISSUE TITLE
Make CC dependency a (default?) feature

### ISSUE THREAD

**Issue by thenextman** [multiformats/rust-multihash#79]
v0.11.3 caused some difficulties for us due to, on some targets, the `blake3` dependency using `cc` to compile intrinsics. 

We maintain a complex cross-platform project that we build for Windows/macOS/Linux/Android/iOS where all C/C++ dependencies are prebuilt and packaged through conan.io packages with strict compilation options. With this in mind, we try to enforce builds where the "cc" crate is never called because of the lack of control over the build process and compilation options. In this specific case, we got an unexpected and sudden build break in our CI (`blake3` is using linker options that are incompatible with ours on Windows x86).

For now, I've pinned 0.11.2 and we can always patch `multihash` ourselves. However, it would be really nice if there was some possibility to just control the hash algorithms available via features. We don't need `blake3` support, and being able to disable it would be more "environmentally friendly". As it stands right now, a C/C++ application built with MSVC for x86 and using /MT won't be able to compile with `multihash` as a dependency. 

Possibly related to #32

**Comment by mriise** [comment]
Are you asking more to have blake3 be an optional feature, or to have specific CC dependencies as the title suggests?

I'm willing to make another PR to setup blake3 as an optional dependency but would prefer to wait to see where the [rewrite](https://github.com/multiformats/rust-multihash/pull/69) is headed and work from there.

**Comment by vmx** [comment]
@mriise The rewrite still takes a while (as I'm too busy with other things). I like your idea of having blake3 as an optional feature, that should unblock @thenextman. I could do a v0.11.4 release with it.

**Comment by thenextman** [comment]
@mriise Having blake3 as an optional feature would unblock things for us and be appreciated, like @vmx said. Ultimately, we can't build C "on the fly" like that - I would expect the same is true for anyone building cross-platform C projects of certain complexity, because there's no "one size fits all" with build processes and compilation options. Sorry if my issue title was not very clear :)

**Comment by mriise** [comment]
> 
> 
> @mriise do you have time to work on making it optional? If not, then I can work on it some time the next few days. I just want to prevent double efforts.

I'll see what i can do, as i'm not terribly familiar with macros.
As a side note, (and from what i currently understand about macros) it would be easy to setup each algorithm (or family) as its own feature, with the default being all, but people could list out the algorithms they need and the others would be excluded.
although that may be just be unneeded feature bloat.

**Comment by vmx** [comment]
> but people could list out the algorithms they need and the others would be excluded.

This is what tiny-multihash is doing (which is what likely will go upstream). Though for now I'd just restrict it to blake3 to unblock @thenextman.

**Comment by mriise** [comment]
As a side note, 
It bothered me that Blake3 was trying to compile C in the first place, after a bit more looking around, it turns out you can force the rust only implementation by using the "pure" feature.
